lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -1560,6 +1572,55 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
         rust!(self.out, "}}");
         rust!(self.out, "}}).collect()");
         rust!(self.out, "}}");
+
+        Ok(())
+    }
+
+    fn emit_expected_tokens_from_states_fn(&mut self) -> io::Result<()> {
+        let parameters = vec![
+            format!(
+                "{p}states: &[{typ}]",
+                p = self.prefix,
+                typ = self.custom.state_type
+            ),
+            format!("_: {}", self.phantom_data_type()),
+        ];
+
+        self.out
+            .fn_header(
+                &Visibility::Priv,
+                format!("{}expected_tokens_from_states", self.prefix),
+            )
+            .with_type_parameters(&self.custom.machine.type_parameters)
+            .with_where_clauses(&self.custom.machine.where_clauses)
+            .with_parameters(parameters)
+            .with_return_type("alloc::vec::Vec<alloc::string::String>")
+            .emit()?;
+
+        rust!(self.out, "{{");
+
+        // Grab any terminals in the current state which would have resulted in a successful parse,
+        // as verified using accepts()
+        rust!(
+            self.out,
+            "{}TERMINAL.iter().enumerate().filter_map(|(index, terminal)| {{",
+            self.prefix,
```

Comment:
```suggestion
            "{p}TERMINAL.iter().enumerate().filter_map(|(index, terminal)| {{",
            p = self.prefix,
```

we generally use a named variable for prefix, even if only used once, I think?

Reply:
meh, not worth waiting for it, we're a bit inconsistent already